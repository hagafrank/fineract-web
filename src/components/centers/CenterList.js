import React from "react";
import PropTypes from "prop-types";
import { FormattedMessage } from "react-intl";

const CenterList = ({ centers }) => (
  <table className="table">
    <thead>
      <tr>
        <th>
          <FormattedMessage id="CenterList.name" />
        </th>
        <th>
          <FormattedMessage id="CenterList.accountNumber" />
        </th>
        <th>
          <FormattedMessage id="CenterList.externalId" />
        </th>
        <th>
          <FormattedMessage id="CenterList.status" />
        </th>
        <th>
          <FormattedMessage id="CenterList.office" />
        </th>
      </tr>
    </thead>
    <tbody>
      {centers.map((center) => {
        return (
          <tr key={center.id}>
            <td>{center.name}</td>
            <td>{center.accountNo}</td>
            <td>{center.externalId}</td>
            <td>{center.status.value}</td>
            <td>{center.officeName}</td>
          </tr>
        );
      })}
    </tbody>
  </table>
);

CenterList.propTypes = {
  centers: PropTypes.array.isRequired
};

export default CenterList;
