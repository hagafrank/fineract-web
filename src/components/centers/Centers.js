import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import CenterList from "./CenterList";
import Spinner from "../common/Spinner";
import { FormattedMessage } from "react-intl";
import { loadCenters } from "../../redux/actions/centerActions";

class Centers extends React.Component {
  constructor() {
    super();

    this.state = {
      loading: false
    };
  }

  toggleSpinner = (show) => {
    this.setState({ loading: show });
  };

  componentDidMount() {
    this.toggleSpinner(true);
    this.props
      .loadCenters()
      .then(() => this.toggleSpinner(false))
      .catch((error) => {
        this.toggleSpinner(false);
        console.error("Loading centers failed. " + error);
      });
  }

  render() {
    return (
      <>
        <h2>
          <FormattedMessage id="Centers.header" />
        </h2>
        {this.state.loading ? (
          <Spinner />
        ) : (
          <CenterList centers={this.props.centers} />
        )}
      </>
    );
  }
}

Centers.propTypes = {
  centers: PropTypes.array.isRequired,
  loadCenters: PropTypes.func.isRequired
};

function mapStateToProps(state) {
  return {
    centers: state.centers.centers || []
  };
}

const mapDispatchToProps = {
  loadCenters
};

export default connect(mapStateToProps, mapDispatchToProps)(Centers);
