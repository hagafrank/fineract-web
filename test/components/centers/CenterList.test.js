import React from "react";
import { centers } from "../../mock_data/centers/centers";
import CenterList from "../../../src/components/centers/CenterList";
import createComponentWithIntl from "../../helpers/jest-snapshot/createComponentWithIntl";

test("centers list should be rendered", () => {
  const component = createComponentWithIntl(<CenterList centers={centers} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
