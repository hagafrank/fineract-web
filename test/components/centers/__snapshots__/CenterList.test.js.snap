// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`centers list should be rendered 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th>
        Name
      </th>
      <th>
        Account #
      </th>
      <th>
        External Id
      </th>
      <th>
        Status
      </th>
      <th>
        Office
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        GrupoBeat
      </td>
      <td>
        000000004
      </td>
      <td />
      <td>
        Active
      </td>
      <td>
        Head Office
      </td>
    </tr>
  </tbody>
</table>
`;
